--- base.c
+++ best.c
@@ -713,8 +713,8 @@
 extern unsigned char far PherMapRN[64][32];
 extern unsigned char far PherMapRT[64][32];
 extern unsigned char far MapA[128][64];
+extern int far fd_50F6_049A;
 extern unsigned char far LifeA[128][64];
-extern int far fd_50F6_049A;
 extern int far fd_50F6_04C2;
 extern int far fd_50F6_0502;
 extern int far fd_50F6_0496;
@@ -737,8 +737,8 @@
     mx = fd_50F6_0508.x + x;
     my = fd_50F6_0508.y + y;
     if (my > 0x3f) {
+        my &= 0x3f;
         mx += 0x40;
-        my &= 0x3f;
     }
     g_94E4 = 0;
     g_9126 = 0;
