--- base.c
+++ best.c
@@ -38,11 +38,11 @@
 extern long far f_171C_1750(void);
 extern long far f_171C_1772(void);
 extern void far WinPrintf(char far *format, ...);
-extern char far * far * far f_171C_1A9E(long size, int kind, char far *name);
+extern char far * far * far f_171C_1A9E(long, int, char far *);
 extern char far * far * far fd_50F6_10E2;
 extern char near g_5A97;
+extern int far fd_50F6_3858;
 extern int far fd_50F6_3856;
-extern int far fd_50F6_3858;
 extern void (far * far fd_50F6_38B8)();
 extern void (far * far fd_50F6_38BC)();
 extern void far o00_3126_0000();
@@ -147,8 +147,8 @@
 extern struct Rect far fd_50F6_10D2;
 extern struct Rect far fd_50F6_38C2;
 extern int far fd_50F6_10DE;
+extern int far fd_50F6_10E0;
 extern int far fd_50F6_38C0;
-extern int far fd_50F6_10E0;
 extern void far f_1CE2_0410(struct Rect far *rect, int width);
 extern void far clip_Pop(void);
 
