--- base.c
+++ best.c
@@ -1943,7 +1943,6 @@
     Handle h;
     int i;
     int row;
-    unsigned n;
 
     f_24AB_02AD(2);
     bpp = 2;
@@ -1982,15 +1981,14 @@
         balTileRect.bottom = ht + balTileRect.top;
         balTileRect.right = balTileRect.left + wt;
         wpix = g_19BE * wt;
-        n = ht * wt * rowbytes;
-        balBufHandle = f_171C_1A9E((long)(n + 4), 9, "balbuf");
+        balBufHandle = f_171C_1A9E((long)(ht * wt * rowbytes + 4), 9, "balbuf");
         balBufPtr = f_171C_1B84(balBufHandle);
         bufp = balBufPtr;
         ((int far *)bufp)[0] = g_19BE * wt;
         ((int far *)bufp)[1] = g_19C0 * ht;
         skip = g_19C0 / div * ((int far *)bufp)[0] - wt * bpp;
         dst = bufp + 4;
-        _fmemset(dst, 0, n);
+        _fmemset(dst, 0, ht * wt * rowbytes);
         balloonRect.left = g_19BE * balTileRect.left + fd_50F6_110C.left;
         balloonRect.top = g_19C0 * balTileRect.top + fd_50F6_110C.top;
         balloonRect.bottom = balloonRect.top + ((int far *)bufp)[1];
