--- base.c
+++ best.c
@@ -62,7 +62,7 @@
 extern Block far * near g_91A8;
 extern Block far * near g_91AC;
 
-extern void far Punt(char far *format, ...);
+extern void far Punt(char far *, ...);
 extern void far WinPrintf(char far *format, ...);
 extern void far f_195A_001D(void);
 extern void far f_195A_0035(void);
@@ -589,6 +589,7 @@
     int compacted;
     Block far *n;
     char buffer[100];
+    int tmp2;
 
     first = 1;
     compacted = 0;
@@ -606,7 +607,8 @@
             if (best->paras > paras + 4) {
                 WinPrintf("<SPLIT>");
                 f_171C_068C(best, best->paras - paras, 0);
-                n = BLK(SEG(best) + best->paras);
+                tmp2 = SEG(best) + best->paras;
+                n = BLK(tmp2);
                 f_171C_068C(n, paras, type);
                 f_171C_0160(best, 1);
                 return n;
@@ -623,7 +625,7 @@
         if (!compacted) {
             WinPrintf("<CL>");
             if (s_2F3E > paras && f_171C_0CF4(0)) {
-                first = compacted = 1;
+                compacted = first = 1;
                 continue;
             }
         }
