--- base.c
+++ best.c
@@ -807,16 +807,16 @@
                 if (g_3DB2 == 320) {
                     t = y * 10 + 12;
                     l = ((x + 6) * 28 - t) / 2 + fd_50F6_10D2.left + 10;
+                    r.bottom = (t + 0x47) / 2 + fd_50F6_10D2.top + 4;
                     r.left = l;
-                    r.bottom = (t + 0x47) / 2 + fd_50F6_10D2.top + 4;
                     r.top = r.bottom - h / 2;
                     r.right = l + 4;
                 } else {
                     l = (x + 6) * 28 + fd_50F6_10D2.left - y * 10;
                     r.left = l;
                     r.bottom = y * 10 + fd_50F6_10D2.top + 0x47;
+                    r.right = l + 9;
                     r.top = r.bottom - h;
-                    r.right = l + 9;
                 }
                 f_1CE2_046D(&r, f_1B4E_000D(15));
             }
@@ -825,8 +825,8 @@
                 if (g_3DB2 == 320) {
                     t = y * 10 + 12;
                     l = (x * 28 - t + 0xb4) / 2 + fd_50F6_10D2.left + 10;
+                    r.bottom = (t + 0x47) / 2 + fd_50F6_10D2.top + 4;
                     r.left = l;
-                    r.bottom = (t + 0x47) / 2 + fd_50F6_10D2.top + 4;
                     r.top = r.bottom - h / 2;
                     r.right = l + 4;
                 } else {
